### ISSUE TITLE
Support ruff for quality checking

### ISSUE THREAD

**Issue by AdamWill** [Bachmann1234/diff_cover#362]
[Ruff](https://github.com/astral-sh/ruff) seems to be the cool new kid on the block for Python linting. Could we have diff-cover support it for quality checking? At a quick look I guess it might be pretty simple if it works nicely with `RegexBasedDriver`, if I get time to try it myself I will do...

**Comment by mattbachmann-toast** [comment]
100% open to a pr on this. I don't typically add features to diff cover myself since I don't actively work on any python projects right now and I would prefer anything that gets added has at least one person who intends on using the feature.

I see my role as maintaining dependencies, keeping python up to date, managing prs that come in and helping with issues when there is enough detail for me to understand and reproduce the problem.

**Comment by Bachmann1234** [comment]
Sorry, responded from a work account before.

100% open to a pr on this. I don't typically add features to diff cover myself since I don't actively work on any python projects right now and I would prefer anything that gets added has at least one person who intends on using the feature.

I see my role as maintaining dependencies, keeping python up to date, managing prs that come in and helping with issues when there is enough detail for me to understand and reproduce the problem.

**Comment by Gary-LePeep** [comment]
I would also like to see this feature developed, I definitely find it useful since ruff is like the all-around linter

**Comment by ssbarnea** [comment]
My team switched to ruff an year ago, ~4 months later we removed flake8 and later we reduces our use of pylint and pydocstyle gradually as each new version of ruff covered extra checks done by these. At some point we will likely remove pylint and pydocstyle too. -- we use [pre-commit hooks ](https://pre-commit.com/) to orchestrate running these.

PS. We already started to use ruff reformatting, but we run black after it too to cover for the very few cases where these diverge.

**Comment by kingbuzzman** [comment]
Expect a preliminary PR soon.

**Comment by kingbuzzman** [comment]
@Bachmann1234 this could be closed

**Comment by Bachmann1234** [comment]
I typically close on release

**Comment by kingbuzzman** [comment]
@Bachmann1234 can we do a release soon? I'd like to use the `ruff` stuff 😇

**Comment by Bachmann1234** [comment]
Yeah I'll do one today.

**Comment by Bachmann1234** [comment]
Try out https://pypi.org/project/diff-cover/9.3.0/ yall! Big shoutout for @kingbuzzman

**Comment by kingbuzzman** [comment]
@fellnerse you're right, ive addressed it [here](https://github.com/Bachmann1234/diff_cover/pull/515)
